gem/interpreter.py

```diff
@@ -274,18 +294,23 @@ def _(e, self):
 def _(e, self):
     """Index sums reduce over the given axis."""
     val = self(e.children[0])
-    idx = val.fids.index(e.index)
-    return Result(val.arr.sum(axis=idx),
-                  val.fids[:idx] + val.fids[idx+1:])
+    idx = tuple(map(val.fids.index, e.multiindex))
+    rfids = tuple(fi for fi in val.fids if fi not in e.multiindex)
+    return Result(val.arr.sum(axis=idx), rfids)
 
 
 @_evaluate.register(gem.ListTensor)  # noqa: F811
 def _(e, self):
     """List tensors just turn into arrays."""
     ops = [self(o) for o in e.children]
-    assert all(ops[0].fids == o.fids for o in ops)
-    return Result(numpy.asarray([o.arr for o in ops]).reshape(e.shape),
-                  ops[0].fids)
+    tmp = Result.empty(*ops)
+    arrs = []
+    for o in ops:
+        arr = numpy.empty(tmp.fshape)
+        arr[:] = o.broadcast(tmp.fids)
+        arrs.append(arr)
+    arrs = numpy.moveaxis(numpy.asarray(arrs), 0, -1).reshape(tmp.fshape + e.shape)
```

Comment:
Yes, in particular, one of its entries might be zero (which loses free indices in GEM, but not in UFL), while others may have free indices.